packages/flutter_tools/bin/podhelper.rb

```diff
@@ -123,6 +131,12 @@ def flutter_additional_macos_build_settings(target)
   end
 
   target.build_configurations.each do |build_configuration|
+    # ARC code targeting macOS 10.10 does not build on Xcode 14.3. Force to at least macOS 10.11.
+    build_configuration.build_settings['MACOSX_DEPLOYMENT_TARGET'] = '10.11' if force_to_arc_supported_min
+
+    # Skip other updates if it's not a Flutter plugin (transitive dependency).
+    next unless target.dependencies.any? { |dependency| dependency.name == 'FlutterMacOS' }
```

Comment:
what if it has dependencies that have dependencies on FlutterMacOS?

Reply:
They would each be processed separately in this loop.  The ones that depend on Flutter (Flutter plugins) will continue no matter if it's a dependency of something else or not, but the ones that don't (non-Flutter plugin transitive dependencies) won't.

Did I understand the question?

Reply:
Ahh, makes sense!